src/back/glsl/features.rs

```diff
@@ -236,8 +238,46 @@ impl<'a, W> Writer<'a, W> {
                         self.features.request(Features::ARRAY_OF_ARRAYS)
                     }
 
+                    // If the array is dynamically sized
                     if size == crate::ArraySize::Dynamic {
-                        self.features.request(Features::DYNAMIC_ARRAY_SIZE);
+                        let mut is_used = false;
+
+                        // Check if this type is used in a global that is needed by the current entrypoint
+                        for (global_handle, global) in self.module.global_variables.iter() {
+                            // Skip unused globals
+                            if ep_info[global_handle].is_empty() {
+                                continue;
+                            }
+
+                            // If this array is the type of a global, then this array is used
+                            if global.ty == ty_handle {
+                                is_used = true;
+                                break;
+                            }
+
+                            // If the type of this global is a struct
+                            if let crate::TypeInner::Struct { ref members, .. } =
+                                self.module.types[global.ty].inner
+                            {
+                                // Check the last element of the struct to see if it's type uses
+                                // this array
+                                if let Some(last) = members.last() {
```

Comment:
it seems to me that we should just compare `last.ty == ty_handle` here instead of matching the type

Reply:
Ah, good point, because we already checked.

Pushed an update.